### PR TITLE
fortio update

### PR DESCRIPTION
**Pull request by ldemailly**
- build and use fortio in perf/standalone scripts
- fix for having more connections than requested
- disabling compression by default (saves cpu), re-enable using new `-compression` flag
- updating sample in README.md to a test using echosrv (slightly lower latency)
- (misc/minor) also changing threshold for sleep time histogram to 5%

### CODE REVIEW

**Comment by ldemailly** [comment]
this fixes 2 entries from #453

**Comment by ldemailly** [comment]
this version now uses the exact/expected number of sockets

it still uses a lot more cpu than wrk but it can now trigger 60k qps (up from 40 but still lower than wrk's 100k qps) out of echosrv

**Comment by ldemailly** [comment]
With #470 it's very close to wrk

File: devel/fortio/http.go [hunk 1]
```diff
@@ -118,7 +132,32 @@ func FetchURL(url string) (int, []byte) {
...
+func NewClient(url string, numConnections int, compression bool) *Client {
+	req := newHTTPRequest(url)
+	if req == nil {
+		return nil
+	}
+	client := Client{
+		url,
+		req,
+		&http.Client{
+			Timeout: 3 * time.Second,
```
Comment (douglas-reid) [inline devel/fortio/http.go]: curious: any special reason for 3s timeouts ?

File: devel/fortio/periodic.go [hunk 2]
```diff
@@ -170,7 +170,7 @@ func (r *periodicRunner) Run() {
 	fmt.Printf("Ended after %v : %d calls. qps=%.5g\n", elapsed, functionDuration.Count, actualQPS)
 	if useQPS {
 		percentNegative := 100. * float64(sleepTime.hdata[0]) / float64(sleepTime.Count)
-		if percentNegative > 3 {
+		if percentNegative > 5 {
```
Comment (douglas-reid) [inline devel/fortio/periodic.go]: worth documenting choice of 5 here?

File: devel/fortio/http.go (hunk 1 above)
Comment (ldemailly) [inline devel/fortio/http.go]: it means (I think) the max duration will be 3000ms before giving up - given we're expecting data in the single digit ms range most of the time, it seemed enough but I don't really have a good reason - probably should make it configurable or not set it

File: devel/fortio/periodic.go (hunk 2 above)
Comment (ldemailly) [inline devel/fortio/periodic.go]: I'll add a comment, thx, I think it's somewhat arbitrary  - it's how often we are falling behind for the target qps - 5% is in the noise (actually maybe 10) - the only effect is more verbose debugging and the warning

**Pull request merged**